ResearchKit/Common/ORKPasscodeStepViewController.m

```diff
@@ -647,4 +674,46 @@ - (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
     return _shouldResignFirstResponder;
 }
 
+- (void)forgotPasscodeTapped
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(forgotPasscodeTapped:)])
+    {
+        [self.passcodeDelegate forgotPasscodeTapped:_forgotPasscodeButton];
+    }
+}
+
+- (BOOL) hasForgotPasscode
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(hasForgotPasscode)])
+    {
+        return [self.passcodeDelegate hasForgotPasscode];
+    }
+    return NO;
+}
+
+- (NSString*) forgotPasscodeButtonText
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(textForForgotPasscode)])
+    {
+        return [self.passcodeDelegate textForForgotPasscode];
+    }
+    return @"";
```

Comment:
Should probably have a default here that is *not* empty string. I would recommend a search and replace on `/* Passcode step */` to add the text for the forgot password to all the string files with English as default - let Apple worry about fixing that. ;)